types/jest/index.d.ts

```diff
@@ -695,7 +695,7 @@ declare namespace jest {
          *   expect(mock).toBeCalledWith(expect.any(Number));
          * });
          */
-        any<T extends Constructor>(classType: T): T extends Func ? ReturnType<T> : InstanceType<T>;
+        any<T>(classType: T): T extends Func ? ReturnType<T> : T extends Constructor ? InstanceType<T> : any;
```

Comment:
I understand the joy of playing with generics, but that is not what `expect.any` returns. The [return type](https://github.com/facebook/jest/blob/a570638d6325dac0a52034ce0d208790af3fcf1a/packages/expect/src/types.ts#L74-L79) of any of asymmetric matchers is:

```ts
type AsymmetricMatcher = {
  asymmetricMatch(other: unknown): boolean;
  toString(): string;
  getExpectedType?(): string;
  toAsymmetricMatcher?(): string;
};
```

If the aim of `@types/jest` is to reflect the reality, current types are obviously wrong. If the aim is different, then please ignore me (;

Reply:
Thanks @mrazauskas. Yes, goal was align with `expect.any` types with that if jest asymmetric matchers. But I understand that it requires `any` types to be more flexible. I have accepted your suggested changes. Please review when you get a chance.